tutorials/managing-gcp-projects-with-terraform/index.md

```diff
@@ -0,0 +1,285 @@
+ services = [
+   "compute-component.googleapis.com"
+ ]
+}
+ 
+output "project_id" {
+ value = "${google_project.project.project_id}"
+}
+```
+ 
+*View the code [on GitHub](./project.tf).*
+
+Terraform resources used:
+
+- [`provider "google"`](https://www.terraform.io/docs/providers/google/index.html): The Google cloud provider config. The credentials will be pulled from the `GOOGLE_CREDENTIALS` environment variable (set later in tutorial).
+- [`resource "random_id"`](https://www.terraform.io/docs/providers/random/r/id.html): Project IDs must be unique. Generate a random one prefixed by the desired project ID. 
+- [`resource "google_project"`](https://www.terraform.io/docs/providers/google/r/google_project.html): The new project to create, bound to the desired organization ID and billing account.
+- [`resource "google_project_services"`](https://www.terraform.io/docs/providers/google/r/google_project_services.html): Services and APIs enabled within the new project. Note that if you visit the web console after running terraform, additional APIs may be implicitly enabled and Terraform would become out of sync. Re-running `terraform plan` will show you these changes before terraform attempts to disable the APIs that were implicitly enabled. You can also set the full set of expected APIs beforehand to avoid the synchronization issue. 
+- [`output "project_id"`](https://www.terraform.io/intro/getting-started/outputs.html): The project ID is randomly generated for uniqueness. Use an output variable to display it after Terraform runs for later reference. 
+ 
+The `compute.tf` file:
+
+[embedmd]:# (compute.tf /data/ //)
+
+```hcl
+data "google_compute_zones" "available" {}
+ 
+resource "google_compute_instance" "default" {
+ project = "${google_project_services.project.project}"
+ zone = "${data.google_compute_zones.available.names[0]}"
+ name = "tf-compute-1"
+ machine_type = "f1-micro"
+ disk {
+   image = "ubuntu-1604-xenial-v20170328"
+ }
+ network_interface {
+   network = "default"
+   access_config {
+   }
+ }
+}
+ 
+output "instance_id" {
+ value = "${google_compute_instance.default.self_link}"
+}
+```
+
+*View the code [on GitHub](./compute.tf).*
+ 
+Terraform resources used:
+
+- [`data "google_compute_zones"`](https://www.terraform.io/docs/providers/google/d/google_compute_zones.html): Data resource used to lookup available Compute Engine zones, bound to the desired region. Avoids hard-coding of zone names.
+- [`resource "google_compute_instance"`](https://www.terraform.io/docs/providers/google/r/compute_instance.html): The compute instance bound to the newly created project. Note that the project is referenced from the `google_project_services.project` resource. This is to tell Terraform to create it after the Compute Engine API has been enabled. Otherwise, Terraform would try to enable Compute Engine API and create the instance at the same time, leading to an attempt to create the instance before the Compute Engine API is fully enabled.
```

Comment:
s/compute instance/Compute Engine instance/

also, s/enable Compute/enable the Compute/